osu.Game/Screens/SelectV2/BeatmapCarouselFilterMatching.cs

```diff
@@ -41,17 +41,20 @@ private IEnumerable<CarouselItem> matchItems(IEnumerable<CarouselItem> items, Fi
             {
                 var beatmap = (BeatmapInfo)item.Model;
 
-                if (checkMatch(beatmap, criteria))
-                {
-                    countMatching++;
-                    yield return item;
-                }
+                if (beatmap.Hidden)
+                    continue;
+
+                if (!checkCriteriaMatch(beatmap, criteria))
+                    continue;
+
+                countMatching++;
+                yield return item;
             }
 
             BeatmapItemsCount = countMatching;
         }
 
-        private static bool checkMatch(BeatmapInfo beatmap, FilterCriteria criteria)
+        private static bool checkCriteriaMatch(BeatmapInfo beatmap, FilterCriteria criteria)
```

Comment:
why was this renamed here?

Reply:
Making it explicit that this method only checks against the filter criteria, i.e. it doesn't check whether the beatmap is hidden (that's done outside).

Reply:
I find it slightly wrong that hidden is being done in this filter in the first place. I'd prefer it be its own filter. But let's just go with this for now because that might be considered nit-picking.